packages/address-book-controller/src/AddressBookController.ts

```diff
@@ -128,15 +127,15 @@ export class AddressBookController extends BaseController<
    * @param name - Nickname to associate with this address.
    * @param chainId - Chain id identifies the current chain.
    * @param memo - User's note about address.
-   * @returns Boolean indicating if the address was successfully set.
    * @param addressType - Contact's address type.
+   * @returns Boolean indicating if the address was successfully set.
    */
   set(
     address: string,
     name: string,
     chainId = '1',
     memo = '',
-    addressType = AddressType.unknown,
+    addressType = undefined as unknown as AddressType,
```

Comment:
We should not need any default here; `undefined` will automatically be the default for unset parameters.

This type cast should not be needed either; this is only causing a TypeScript error because this property remains mandatory for the `AddressBookEntry` type.

Here is how we can mark this parameter as optional:

```suggestion
    addressType?: AddressType,
```

The `AddressBookEntry` interface will need to be updated above as well, to make the `addressType` property optional, e.g.

```
AddressBookEntry {
  address: string;
  name: string;
  chainId: string;
  memo: string;
  isEns: boolean;
  addressType?: AddressType;
}
```

Reply:
Done, thank you for the suggestion.